pkg/controller/discord/new.go

```diff
@@ -335,3 +336,60 @@ func (c *controller) GetOgifLeaderboard(ctx context.Context, after time.Time, li
 
 	return leaderboard, nil
 }
+
+func (c *controller) ListDiscordChannelMessageLogs(ctx context.Context, channelID string, startTime *time.Time, endTime *time.Time) ([]model.DiscordTextMessageLog, error) {
+	threads, err := c.service.Discord.ListActiveThreadsByChannelID(c.config.Discord.IDs.DwarvesGuild, channelID)
+	if err != nil {
+		return nil, err
+	}
+	channelIDs := make([]string, 0)
+	channelIDs = append(channelIDs, channelID)
+	for _, thread := range threads {
+		channelIDs = append(channelIDs, thread.ID)
+	}
+
+	members, err := c.service.Discord.GetMembers()
+	if err != nil {
+		return nil, err
+	}
+
+	membersMap := make(map[string]string)
+	for _, mem := range members {
+		membersMap[fmt.Sprintf("<@%s>", mem.User.ID)] = fmt.Sprintf("@%s", mem.User.Username)
+	}
+
+	messageLogs := make([]model.DiscordTextMessageLog, 0)
+
+	for _, channnelID := range channelIDs {
+		messages, err := c.service.Discord.GetChannelMessagesInDateRange(channnelID, 100, startTime, endTime)
+		if err != nil {
+			return nil, err
+		}
+
+		for _, msg := range messages {
+			messageLogs = append(messageLogs, model.DiscordTextMessageLog{
```

Comment:
@namnhce @nnhuyhoang 
this seems like a potential OOM, since on prod we only have 100MB-ish memory, make sure we dont OOM here

Reply:
will test on dev with similar amount of message, if cannot handle them , find another way